code/HabitTracker/app/src/main/java/com/example/habittracker/activities/HabitViewActivity.java

```diff
@@ -156,6 +167,7 @@ public void onClick(View view) {
                 Intent intent = new Intent(getApplicationContext(),ProgressTrackingActivity.class);
                 Bundle bundle = new Bundle();
                 bundle.putSerializable("habit", habit);
+                bundle.putString("user",SharedInfo.getInstance().getCurrentUser().getUsername());
```

Comment:
May I ask, why are you passing the current user int he bundle not just getting it on the progress activity? (Curious)

Reply:
Removed